components/datetime/README.md

```diff
@@ -69,15 +69,15 @@ let date = typed_date.to_iso().to_any();
 let formatted_date = dtf.format(&date).expect("Calendars should match");
 let typed_formatted_date = typed_dtf.format(&typed_date);
 
-assert_writeable_eq!(formatted_date, "Sep 12, 2020, 12:34 PM");
-assert_writeable_eq!(typed_formatted_date, "Sep 12, 2020, 12:34 PM");
+assert_writeable_eq!(formatted_date, "Sep 12, 2020, 12:34\u{202f}PM");
```

Comment:
Question/Discussion: Do we want to put `\u{202f}` in our docs, or a real NNBSP?

Reply:
In docs a user probably doesn't care about \u{202f} so a real NNBSP would be more readable for quickly understanding how to use the code.

Reply:
@robertbastian agrees with @gregtatum and says the general rule should be to render literal Unicode unless it is confusable

@Manishearth says that we should clarify with a non-rendering docs comment so that future people updating the example know what to expect

@younies says it's weird to see the code point sequences in docs; why would you want to do that?

@echeran says to read his comment above to answer that question, but the assertion made by @robertbastian is a bit strict (e.g. combining marks)

@sffc says that unit tests should use the escape sequence because the audience is i18n engineers who care about the difference

@robertbastian says that modern editors don't have problems with Unicode and we should strive to make unit tests readable

@younies says that it's useful in writing unit tests to be more explicit

@echeran says that modern editors can have problems with unicode; it doesn't work great in Eclipse, Intellij, etc. There's an example of how they screw up grapheme clusters.

@Manishearth says that it's important for invisible and whitespace characters to be explicit because when editing tests it's easy to miss it. For visible characters, it seems fine to leave it in the native writing system.

Conclusion:

- For docs, render most characters, including whitespace
- Elsewhere (in code and unit tests), use explicit escapes for whitespace and invisible characters, and explicitly render visible characters